### PR TITLE
refactor: extract message handlers and use pattern matching in sync

### PR DESCRIPTION
**Pull request by matthiasn**
## Summary by CodeRabbit

* **Bug Fixes**
  * Improved resilience in sync processing with better error handling and fallback to avoid orphaned entry references and reduce retry storms.

* **Improvements**
  * Added sequence-gap detection and logging for improved sync reliability monitoring.
  * Increased diagnostics and duplication-avoidance logging for clearer observability.

* **Refactor**
  * Consolidated per-message handling and outbox enqueue flow for clearer, more reliable processing and merge behavior.

* **Tests**
  * Added extensive edge-case tests for outbox/enqueue behavior.

* **Chores**
  * Release version and changelog updated.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: lib/features/sync/outbox/outbox_service.dart [hunk 1]
```diff
@@ -241,482 +241,127 @@ class OutboxService {
...
+            );
+            _loggingService.captureEvent(
+              'enqueue type=SyncBackfillResponse hostId=${msg.hostId} '
+              'counter=${msg.counter} deleted=${msg.deleted}',
+              domain: 'OUTBOX',
+              subDomain: 'enqueueMessage',
+            );
+            return false;
+          }(),
+      };
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply lib/features/sync/outbox/outbox_service.dart:301]: done
